src/features/lightspeed/lightSpeedOAuthProvider.ts

```diff
@@ -349,7 +353,7 @@ export class LightSpeedAuthenticationProvider
       client_id: LIGHTSPEED_CLIENT_ID,
       code: code,
       code_verifier: CODE_VERIFIER,
-      redirect_uri: this.redirectUri,
+      redirect_uri: this._externalRedirectUri,
```

Comment:
Here the redirect URI is used again. However, it is not used for redirection and need to be the same value used for redirection to send the authorization code. So `this.setExternalRedirectUri()` is NOT called there and the stored `this._externalRedirectUri` value is set to the `redirect_uri` property.